### PR TITLE
[ao] Standardized InputWeightEqualizationDetector output to single level

### PR DESCRIPTION
**Pull request by vspenubarthi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81673
* #81647
* #81589
* #81523
* #81588
* #81587
* __->__ #81586
* #81585

Summary: Currently the InputWeightEqualizationDetector has a
multi-layered output.

Example
```
{'block1.linear': {'channel_axis_selected': 1,
                   'channel_comparison_metrics': tensor([0.8736, 0.6594, 0.2916], grad_fn=<DivBackward0>),
                   'input_range_info': {'global_max': tensor(9.),
                                        'global_min': tensor(-10.),
                                        'per_channel_max': tensor([9., 9., 9.]),
                                        'per_channel_min': tensor([-10., -10., -10.])},
                   'input_weight_equalization_recommended': [True,
                                                             False,
                                                             False],
                   'threshold': 0.8,
                   'weight_range_info': {'global_max': tensor(0.5618, grad_fn=<UnbindBackward0>),
                                         'global_min': tensor(-0.2211, grad_fn=<UnbindBackward0>),
                                         'per_channel_max': tensor([0.3764, 0.5618, 0.2894], grad_fn=<NotImplemented>),
                                         'per_channel_min': tensor([-0.2211,  0.2213,  0.2228], grad_fn=<NotImplemented>)}},
}
```

With all the levels, it can be hard to parse the information for
anything, especially the planned visualization feature where the data
has to be reorganized. Therefore, to make it standardized across all
detectors, all outputs will be limited to one level.

The new format is:
```
{'block1.linear': { 'channel_axis_selected': 1,
                    'channel_comparison_metrics': tensor([0.5705, 0.9457, 0.8891], grad_fn=<DivBackward0>),
                    'activation_global_max': tensor(9.),
                    'activation_global_min': tensor(-10.),
                    'activation_per_channel_max': tensor([9., 9., 9.]),
                    'activation_per_channel_min': tensor([-10., -10., -10.]),
                    'input_weight_equalization_recommended': [False, True, True],
                    'threshold': 0.8,
                    'weight_global_max': tensor(0.4258, grad_fn=<UnbindBackward0>),
                    'weight_global_min': tensor(-0.4958, grad_fn=<UnbindBackward0>),
                    'weight_per_channel_max': tensor([0.1482, 0.3285, 0.4258], grad_fn=<NotImplemented>),
                    'weight_per_channel_min': tensor([-0.1517, -0.4958, -0.3027], grad_fn=<NotImplemented>)},
}
```

The README will also be updated to reflect this change.

Test Plan: python test/test_quantization.py TestFxDetectInputWeightEqualization

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by HDCharles** [comment]
should give example in both old/new format

**Comment by jerryzh168** [comment]
`activation_global_max`: should we use `input_activation_global_max`? same for other keys related to `activation`
also for `input_weight_equalization_recommended` what are the three booleans mean? should we split this into multiple keys to make it clearer?

**Comment by vspenubarthi** [comment]
@jerryzh168 
yeah, we could use `input_activation_global_max` will make that change.

As for the multiple values, it is a per channel recommendation which is why there are three values, and it doesn't make sense to have a different key per channel. 

This PR should have updated keys: https://github.com/pytorch/pytorch/pull/81587

I would switch the order around but because of the amount of renaming running into a lot of issues, so will address in that next PR

**Review by jerryzh168 — APPROVED** [review]
sg
